### PR TITLE
support setting sticky cookie max age

### LINKED ISSUE #1939: Configurable max-age for cookies
### Do you want to request a *feature* or report a *bug*?

feature

### What did you expect to see?

nginx allows to set name and max-age for sticky cookie

```
upstream backend {
    server webserver1;
    server webserver2;

    sticky cookie srv_id expires=1h domain=.example.com path=/;
}
```

Would be great if we can do it in traefik

### What did you see instead?

Currently max-age hardcoded to `Session`

### Output of `traefik version`: (_What version of Traefik are you using?_)
Version:      v1.3.5
Codename:     raclette
Go version:   go1.8.3
Built:        2017-08-01_04:18:59PM
OS/Arch:      linux/amd64

--------------
P.S. Thank you for this awesome project <3

### PR DESCRIPTION
**Pull request by Patrick0308**
### What does this PR do?

Close #1939 

### More

- [ ] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

**Review by rtribotte — CHANGES_REQUESTED** [review]
Hello @Patrick0308,

Thanks for this contribution!

At a glance, there are a few things missing from the changes.

Whenever a new option is introduced in the dynamic configuration, the CRDs for the KubernetesCRD provider need to be regenerated.
By executing the following command they would be regenerated:
```bash
make generated-crd
```

Also, the dynamic configuration docs references need to be updated:
- adds manually the new option in `docs/content/reference/dynamic-configuration/file.toml` and in `docs/content/reference/dynamic-configuration/file.yaml`
- then run `go generate` to get the other providers dynamic configuration references updated automatically.

**Comment by Patrick0308** [comment]
@rtribotte PTAL

File: docs/content/reference/dynamic-configuration/file.toml [hunk 1]
```diff
@@ -45,6 +45,7 @@
             secure = true
             httpOnly = true
             sameSite = "foobar"
+            maxAge = 1000
```
Comment (rtribotte) [inline docs/content/reference/dynamic-configuration/file.toml:48]:
```suggestion
            maxAge = 42
```

File: docs/content/reference/dynamic-configuration/file.toml [hunk 2]
```diff
@@ -98,6 +99,7 @@
...
```
Comment (rtribotte) [inline docs/content/reference/dynamic-configuration/file.toml:102]:
```suggestion
            maxAge = 42
```

File: docs/content/reference/dynamic-configuration/file.yaml [hunk 3]
```diff
@@ -53,6 +53,7 @@ http:
             secure: true
             httpOnly: true
             sameSite: foobar
+            maxAge: 1000
```
Comment (rtribotte) [inline docs/content/reference/dynamic-configuration/file.yaml:56]:
```suggestion
            maxAge: 42
```

File: docs/content/reference/dynamic-configuration/file.yaml [hunk 4]
```diff
@@ -98,6 +99,7 @@ http:
...
```
Comment (rtribotte) [inline docs/content/reference/dynamic-configuration/file.yaml:102]:
```suggestion
            maxAge: 42
```

File: docs/content/reference/dynamic-configuration/kv-ref.md [hunk 5]
```diff
@@ -221,6 +221,7 @@
 | `traefik/http/services/Service01/loadBalancer/servers/1/url` | `foobar` |
 | `traefik/http/services/Service01/loadBalancer/serversTransport` | `foobar` |
 | `traefik/http/services/Service01/loadBalancer/sticky/cookie/httpOnly` | `true` |
+| `traefik/http/services/Service01/loadBalancer/sticky/cookie/maxAge` | `1000` |
```
Comment (rtribotte) [inline docs/content/reference/dynamic-configuration/kv-ref.md:224]:
```suggestion
| `traefik/http/services/Service01/loadBalancer/sticky/cookie/maxAge` | `42` |
```

File: docs/content/reference/dynamic-configuration/kv-ref.md [hunk 6]
```diff
@@ -237,6 +238,7 @@
 | `traefik/http/services/Service03/weighted/services/1/name` | `foobar` |
 | `traefik/http/services/Service03/weighted/services/1/weight` | `42` |
 | `traefik/http/services/Service03/weighted/sticky/cookie/httpOnly` | `true` |
+| `traefik/http/services/Service03/weighted/sticky/cookie/maxAge` | `1000` |
```
Comment (rtribotte) [inline docs/content/reference/dynamic-configuration/kv-ref.md:241]:
```suggestion
| `traefik/http/services/Service03/weighted/sticky/cookie/maxAge` | `42` |
```

**Review by rtribotte — CHANGES_REQUESTED** [review]
Hello @Patrick0308,

Thanks for the recent changes,

I have a few suggestions, this is nitpicking but we are used to using "42" value for numbers in reference files. This is just a convention, but it is better to remain consistent.

**Comment by Patrick0308** [comment]
@rtribotte PTAL

**Comment by ldez** [comment]
The PR should be rebased on the v3.0 branch

**Pull request closed without merging**